### PR TITLE
PLAT-10368: Stream service

### PR DESCRIPTION
**Pull request by symphony-elias**
### Ticket
PLAT-10368

### Description
Implemented stream endpoints.

### Checklist
- [x] Referenced a ticket in the PR title and in the corresponding section
- [x] Filled properly the description and dependencies, if any
- [x] Unit tests updated or added
- [x] Docstrings added or updated
- [x] Updated the documentation in [docs folder](../docs)

### CODE REVIEW

File: examples/core/service/stream_main.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+from symphony.bdk.gen.pod_model.room_tag import RoomTag
+from symphony.bdk.gen.pod_model.stream_filter import StreamFilter
+from symphony.bdk.gen.pod_model.stream_type import StreamType
+from symphony.bdk.gen.pod_model.user_id import UserId
+from symphony.bdk.gen.pod_model.v2_admin_stream_filter import V2AdminStreamFilter
+from symphony.bdk.gen.pod_model.v2_room_search_criteria import V2RoomSearchCriteria
+from symphony.bdk.gen.pod_model.v3_room_attributes import V3RoomAttributes
+
+
+class StreamMain:
```
Comment (symphony-youri) [inline examples/core/service/stream_main.py]: oh oh that is going to be fun to merge :)

File: symphony/bdk/core/service/stream/stream_service.py [hunk 2]
```diff
@@ -0,0 +1,263 @@
...
+        Wraps the `Demote Owner <https://developers.symphony.com/restapi/reference#demote-owner>`_ endpoint.
+
+        :param user_id: the id of the room owner to be demoted.
+        :param room_id: the room id.
+        :return: None
+        """
+        await self._room_membership_api.v1_room_id_membership_demote_owner_post(
+            id=room_id, payload=UserId(id=user_id), session_token=await self._auth_session.session_token)
+
+    async def create_im_or_mim(self, user_ids: [int]) -> Stream:
```
Comment (symphony-youri) [inline symphony/bdk/core/service/stream/stream_service.py:125]: in the java one we just called it create i think
and we also have create_room below

i'm not sure how you do proper method overloading in python, if you tend to name methods differently or just put all the possible parameters and document them

**Comment by symphony-youri** [comment]
Will we provide something for pagination (list all) later?

File: docs/stream_service.md [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+        bdk_config = BdkConfigLoader.load_from_file("path/to/config.yaml")
+
+        async with SymphonyBdk(bdk_config) as bdk:
+            stream_service = bdk.streams()
+            stream = await stream_service.get_stream("stream_id")
+
+
+if __name__ == "__main__":
+    asyncio.run(MessageMain.run())
```
Comment (symphony-hong) [inline docs/stream_service.md]: it should be `StreamMain` instead

File: symphony/bdk/core/service/stream/stream_service.py [hunk 4]
```diff
@@ -0,0 +1,263 @@
...
+
+        :param user_id: the id of the user to be added to the room.
+        :param room_id: the id of the room in which to add the user.
+        :return: None
+        """
+        await self._room_membership_api.v1_room_id_membership_add_post(
+            payload=UserId(id=user_id), id=room_id,
+            session_token=await self._auth_session.session_token)
+
+    async def remove_member_from_room(self, user_id: int, room_id: int):
```
Comment (symphony-hong) [inline symphony/bdk/core/service/stream/stream_service.py]: `room_id` should be in `str` type

File: tests/core/service/stream/stream_service_test.py [hunk 5]
```diff
@@ -0,0 +1,208 @@
...
+
+
+@pytest.fixture()
+def stream_service(mocked_api_client, auth_session):
+    return StreamService(StreamsApi(mocked_api_client), RoomMembershipApi(mocked_api_client),
+                         ShareApi(mocked_api_client), auth_session)
+
+
+@pytest.mark.asyncio
+async def test_get_stream(mocked_api_client, stream_service):
```
Comment (symphony-hong) [inline tests/core/service/stream/stream_service_test.py]: you can even mock the `StreamsApi`, `RoomMembershipApi` and `ShareApi` to assert if the method was called with the correct params. For instance: https://github.com/SymphonyPlatformSolutions/symphony-api-client-python/blob/2.0/tests/core/service/user/user_service_test.py#L77-L82

File: symphony/bdk/core/service/stream/stream_service.py (hunk 2 above)
Comment (symphony-elias) [reply symphony/bdk/core/service/stream/stream_service.py:125]: There is no such thing as method overloading in python. So either we have two different method names (as I did) or we put one function with all params and discriminate based on which param is not None (which I don't like)

File: docs/stream_service.md (hunk 3 above)
Comment (symphony-elias) [reply docs/stream_service.md:43]: > Will we provide something for pagination (list all) later?

Opened ticket for later: https://perzoinc.atlassian.net/browse/PLAT-10489

File: symphony/bdk/core/service/stream/stream_service.py (hunk 2 above)
Comment (symphony-hong) [reply symphony/bdk/core/service/stream/stream_service.py:125]: I would prefer the first option to make them explicit 🤔

**Comment by symphony-elias** [comment]
> Will we provide something for pagination (list all) later?

I opened https://perzoinc.atlassian.net/browse/PLAT-10489 for a later implementation.

**Review by symphony-hong — APPROVED** [review]
LGTM

**Pull request merged**